interop-tests/src/test/java/example/myapp/helloworld/grpc/RichErrorNativeImpl.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (C) 2018-2021 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package example.myapp.helloworld.grpc;
+
+import akka.NotUsed;
+import akka.grpc.GrpcServiceException;
+import akka.stream.javadsl.Source;
+import com.google.rpc.Code;
+import com.google.rpc.error_details.LocalizedMessage;
+import scala.collection.JavaConverters;
+import scala.collection.SeqLike;
+import scalapb.GeneratedMessage;
+
+import java.util.ArrayList;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+
+public class RichErrorNativeImpl implements GreeterService {
+
+    // #rich_error_model_unary
+    @Override
+    public CompletionStage<HelloReply> sayHello(HelloRequest in) {
+
+        ArrayList<scalapb.GeneratedMessage> ar = new ArrayList<>();
+        ar.add(LocalizedMessage.of("EN", "The password!"));
+
+        GrpcServiceException exception = GrpcServiceException.apply(
+                Code.INVALID_ARGUMENT,
+                "What is wrong?",
+                JavaConverters.asScalaBuffer(ar).toSeq()
```

Comment:
I'm sure we could do a nicer Java API, but perhaps we can leave that for a separate PR